### PR TITLE
Support for ArgMaxOp.

### PR DESCRIPTION
**Pull request by ramakota-helprack**
* Add ArgMax operation with `dims` and `keep_dims`
* Ensure type conversion from i64 type to si64 to avoid issues in `torch-mlir`
* LIT tests for ArgMax

### CODE REVIEW

File: src/Conversion/ONNXToTorch/NN/ArgmaxOp.cpp [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+  LogicalResult matchAndRewrite(Operation *op, ArrayRef<Value> operands,
+      ConversionPatternRewriter &rewriter) const final {
+
+    Location loc = op->getLoc();
+    mlir::MLIRContext *context =  op->getContext();
+    ONNXArgMaxOp op1 = llvm::dyn_cast<ONNXArgMaxOp>(op);
+
+    auto axis 		= op1.axisAttr();       // ::mlir::IntegerAttr
+    auto keepdims	= op1.keepdimsAttr();	// ::mlir::IntegerAttr
```
[the remark this answers is not in the document]
Comment (ramakota-helprack) [reply src/Conversion/ONNXToTorch/NN/ArgmaxOp.cpp]: Fixed

File: test/mlir/conversion/onnx_to_torch/argmax/argmax_select_last_index_0.onnx.mlir [hunk 2]
```diff
@@ -0,0 +1,10 @@
+//RUN: onnx-mlir --EmitONNXIR --run-torch-pass %s -o=%t >/dev/null && cat %t.onnx.mlir | FileCheck -v %s
+module attributes {}  {
+  func @main_graph(%arg0: tensor<4x5xf32>) -> tensor<4xi64> attributes {input_names = ["input"], output_names = ["output"]} {
+//CHECK: %[[DIM:.*]] = torch.constant.int 1
+//CHECK: %[[TRUE:.*]] = torch.constant.bool true
+//CHECK: torch.aten.argmax %arg0, %[[DIM]], %[[TRUE]] : !torch.vtensor<[4,5],f32>, !torch.int, !torch.bool -> !torch.vtensor<[4],si64>
+    %0 = "onnx.ArgMax"(%arg0) {axis = 1 : si64, keepdims = 0 : si64, onnx_node_name = "ArgMax_0", select_last_index = 0 : si64} : (tensor<4x5xf32>) -> tensor<4xi64>
```
[the remark this answers is not in the document]
Comment (ramakota-helprack) [reply test/mlir/conversion/onnx_to_torch/argmax/argmax_select_last_index_0.onnx.mlir:7]: Fixed

[the remark this answers is not in the document]
Comment (ramakota-helprack) [reply test/mlir/conversion/onnx_to_torch/argmax/argmax_select_last_index_0.onnx.mlir:7]: Fixed keepdims issue.

File: test/mlir/conversion/onnx_to_torch/argmax/argmax_keepdim_0.onnx.mlir [hunk 3]
```diff
@@ -0,0 +1,10 @@
+//RUN: onnx-mlir --EmitONNXIR --run-torch-pass %s -o=%t >/dev/null && cat %t.onnx.mlir | FileCheck -v %s
+module attributes {}  {
+  func @main_graph(%arg0: tensor<4x5xf32>) -> tensor<5xi64> attributes {input_names = ["input"], output_names = ["output"]} {
+//CHECK: %[[DIM:.*]] = torch.constant.int 0
+//CHECK: %[[TRUE:.*]] = torch.constant.bool true
+//CHECK: torch.aten.argmax %arg0, %[[DIM]], %[[TRUE]] : !torch.vtensor<[4,5],f32>, !torch.int, !torch.bool -> !torch.vtensor<[5],si64>
```
[the remark this answers is not in the document]
Comment (ramakota-helprack) [reply test/mlir/conversion/onnx_to_torch/argmax/argmax_keepdim_0.onnx.mlir]: Fixed the keepdims issue. Now keep_dims is false with this test.

File: src/Conversion/ONNXToTorch/NN/ArgmaxOp.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (BraunPhilipp) [reply src/Conversion/ONNXToTorch/NN/ArgmaxOp.cpp]: Thanks!

**Review by BraunPhilipp — APPROVED** [review]
LGTM

**Review by ljfitz — APPROVED** [review]

**Pull request merged**